pkg/crc/preflight/preflight_linux.go

```diff
@@ -106,18 +125,72 @@ var libvirtPreflightChecks = [...]Check{
 		cleanupDescription: "Removing /etc/NetworkManager/dnsmasq.d/crc.conf file",
 		cleanup:            removeCrcDnsmasqConfigFile,
 	},
-	{
-		cleanupDescription: "Removing the crc VM if exists",
-		cleanup:            removeCrcVM,
-		flags:              CleanUpOnly,
-	},
 }
 
-func getPreflightChecks(experimentalFeatures bool) []Check {
-	checks := []Check{}
+var vsockPreflightChecks = Check{
+	configKeySuffix:  "check-vsock",
+	checkDescription: "Checking if vsock is correctly configured",
+	check:            checkVsock,
+	fixDescription:   "Checking if vsock is correctly configured",
+	fix:              fixVsock,
+}
+
+func checkVsock() error {
+	info, err := os.Stat("/dev/vsock")
+	if err != nil {
+		return err
+	}
+	if stat, ok := info.Sys().(*syscall.Stat_t); ok {
+		group, err := user.LookupGroupId(fmt.Sprint(stat.Gid))
+		if err != nil {
+			return err
+		}
+		if group.Name != "libvirt" {
+			return errors.New("/dev/vsock is not is the right group")
+		}
+	} else {
+		return errors.New("cannot cast info")
+	}
+	if info.Mode()&0060 == 0 {
+		return errors.New("/dev/vsock doesn't have the right permissions")
+	}
+	return nil
+}
+
+func fixVsock() error {
+	_, _, err := crcos.RunWithPrivilege("modprobe vhost_vsock", "modprobe", "vhost_vsock")
+	if err != nil {
+		return err
+	}
+	_, _, err = crcos.RunWithPrivilege("chown /dev/vsock", "chown", "root:libvirt", "/dev/vsock")
```

Comment:
Fwiw, this is a bit confusing, even if we set the group to `libvirt`, this has nothing to do with libvirt, all that we want is that the user running `crc daemon` can read/write `/dev/vsock`. Since another preflight check ensures this user is part of the `libvirt` group, this ends up doing the right thing. But the group name could be anything, the libvirt daemon/library/... does not care about the `/dev/vsock` permissions.

Reply:
Yes totally true. I reused the group libvirt but I can pick something else. Is there an other good group ? Should I create a new one like vsock ?